public/modules/custom/helfi_ptv_integration/src/HelfiPTV.php

```diff
@@ -0,0 +1,345 @@
+
+      $state->set($key, $city['code']);
+    }
+  }
+
+  private function getAddressData($addressesData)
+  {
+    $nodeData = [];
+    foreach ($addressesData as $address) {
+      if ($address->type === 'Location') {
+        if (empty($address->streetAddress->street)) {
+          continue;
+        }
+        foreach ($address->streetAddress->street as $street) {
+          $nodeData['visiting'][$street->language]['address'] = $street->value . ' ' . $address->streetAddress->streetNumber;
+        }
+        foreach ($address->streetAddress->postOffice as $post) {
+          $nodeData['visiting'][$post->language]['city'] = $address->streetAddress->postalCode . ' ' . $post->value;
+        }
+        if (!empty($address->streetAddress->additionalInformation)) {
+          foreach ($address->streetAddress->additionalInformation as $additional) {
+            $nodeData['visiting'][$additional->language]['additional'] = $additional->value;
+          }
+        }
+      }
+      if ($address->type === 'Postal') {
+        if (empty($address->postOfficeBoxAddress->postOfficeBox)) {
+          continue;
+        }
+        foreach ($address->postOfficeBoxAddress->postOfficeBox as $poBox) {
+          $nodeData['postal'][$poBox->language]['poBox'] = $poBox->value;
+        }
+        foreach ($address->postOfficeBoxAddress->postOffice as $post) {
+          if (!isset($nodeData['postal'][$post->language]['poBox'])) {
+            $nodeData['postal'][$post->language]['poBox'] = $nodeData['postal']['fi']['poBox'];
+          }
+          $nodeData['postal'][$post->language]['city'] = $address->postOfficeBoxAddress->postalCode . ' ' . $post->value;
+        }
+        if (!empty($address->postOfficeBoxAddress->additionalInformation)) {
+          foreach ($address->postOfficeBoxAddress->additionalInformation as $additional) {
+            $nodeData['postal'][$additional->language]['additional'] = $additional->value;
+          }
+        }
+      }
+    }
+    return $nodeData;
+  }
+
+  private function getServiceHours($serviceHoursArray) {
+    foreach ($serviceHoursArray as $serviceHours) {
+      if(empty($serviceHours) || $serviceHours->serviceHourType !== 'DaysOfTheWeek') {
+       continue;
+      }
+      $hours = [];
+      $hours['sv']['additional'] = '';
+      $hours['fi']['additional'] = '';
+      $hours['en']['additional'] = '';
+      if (!empty($serviceHours->additionalInformation)) {
+        foreach ($serviceHours->additionalInformation as $additional) {
+          $hours[$additional->language]['additional'] = $additional->value;
+        }
+      }
+      $hours['fi']['hours'] = '';
+      $hours['en']['hours'] = '';
+      $hours['sv']['hours'] = '';
+      foreach ($serviceHours->openingHour as $day) {
+
+        switch ($day->dayFrom) {
+          case $day->dayFrom === 'Monday':
+            $hours['fi']['hours'] .= 'Ma ' . substr($day->from, 0, -3) . '-' . substr($day->to, 0, -3) . ' ';
```

Comment:
could `substr($day->from, 0, -3) . '-' . substr($day->to, 0, -3) . ' ';` be a function? it would be easier to change if needed.